src/utils/migrateSettings.ts

```diff
@@ -3,6 +3,9 @@ import * as path from "path"
 import * as fs from "fs/promises"
 import { fileExistsAtPath } from "./fs"
 import { GlobalFileNames } from "../shared/globalFileNames"
+import * as yaml from "yaml"
+
+const deprecatedCustomModesJSONFilename = "custom_modes.json"
```

Comment:
Are you really going to use the file _only_ for custom modes? Since you've started the migration process, you might as well call it by a more general name. I wrote my thoughts [here](https://github.com/RooVetGit/Roo-Code/pull/3711#issuecomment-2893345265).

Reply:
I was just thinking of this as changing the format of the custom_modes.json to start, and that changes to the schema and meaning of the file could come later.